src/main/java/stringaddcalculator/StringAddCalculator.java

```diff
@@ -0,0 +1,62 @@
+package stringaddcalculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private StringAddCalculator() {}
+
+    public static int splitAndSum(String input) {
+        if (isNullOrEmpty(input))
+            return 0;
+
+        if (isOneNumber(input))
+            return Integer.parseInt(input);
+
+        if (isCustomSeparator(input))
+            return findSeparatorAndSum(input);
+
+        return getBasicSum(",|:", input);
+    }
+
+    private static boolean isNullOrEmpty(String input) {
+        if (input == null) return true;
+        return input.isEmpty();
+    }
+
+    private static boolean isOneNumber(String input) {
+        return input.length() == 1;
+    }
+
+    private static boolean isCustomSeparator(String input) {
+        return input.startsWith("//");
+    }
+
+    private static int findSeparatorAndSum(String input) {
+        Pattern pattern = Pattern.compile("//(.)\n(.*)");
+        Matcher matcher = pattern.matcher(input);
+        String separator = ",|:";
+        if (matcher.find()) {
+            separator = matcher.group(1);
+            input = matcher.group(2);
+        }
+        return getBasicSum(separator, input);
+    }
+
+    private static int getBasicSum(String separator, String input) {
```

Comment:
`getBasicSum()`이라고 하기에는 separator를 매개변수로 받아, 기본 덧셈 로직이라고 보기는 개인적으로 조금 어색하네요...
```suggestion
    private static int sum(String separator, String input) {
```
살포시 제안드립니다:)